doc/source/whatsnew/v0.25.0.rst

```diff
@@ -305,7 +305,7 @@ Conversion
 ^^^^^^^^^^
 
 - Bug in :func:`DataFrame.astype()` when passing a dict of columns and types the `errors` parameter was ignored. (:issue:`25905`)
--
+- Bug in constructing :class:`DataFrame` from dict, where the index would be sorted instead of using the dict insertion order. (:issue:`24859`)
```

Comment:
Would be preferable if you could reference the method here instead of just the class

Reply:
Would I say `DataFrame.__init__()`, in that case?

Reply:
`:meth:DataFrame.from_dict`

Reply:
The "root" cause of the bug is in `DataFrame.__init__({ some dictionary })`, though.

Reply:
Can also say DataFrame construction then